binaries/cli/src/command/inspect/top.rs

```diff
@@ -0,0 +1,392 @@
+                    pid = Some(process_pid.as_u32());
+                    cpu_usage = process.cpu_usage();
+                    memory_mb = process.memory() as f64 / 1024.0 / 1024.0;
+                    break;
+                }
+            }
+
+            self.node_stats.push(NodeStats {
+                dataflow_id: node_info.dataflow_id,
+                dataflow_name: node_info.dataflow_name.unwrap_or_else(|| "<unnamed>".to_string()),
+                node_id: node_info.node_id,
+                pid,
+                cpu_usage,
+                memory_mb,
+            });
+        }
+
+        self.sort();
+    }
+}
+
+fn run_app<B: Backend>(
+    terminal: &mut Terminal<B>,
+    coordinator_addr: std::net::IpAddr,
+    coordinator_port: u16,
+    refresh_duration: Duration,
+) -> eyre::Result<()> {
+    let mut app = App::new();
+    let mut last_update = Instant::now();
+    let mut system = System::new_all();
+
+    loop {
+        terminal.draw(|f| ui(f, &mut app, refresh_duration))?;
+
+        let timeout = refresh_duration
+            .checked_sub(last_update.elapsed())
+            .unwrap_or(Duration::from_millis(100));
+
+        if event::poll(timeout)? {
+            if let Event::Key(key) = event::read()? {
+                if key.kind == KeyEventKind::Press {
+                    match key.code {
+                        KeyCode::Char('q') | KeyCode::Esc => {
+                            return Ok(());
+                        }
+                        KeyCode::Down | KeyCode::Char('j') => {
+                            app.next();
+                        }
+                        KeyCode::Up | KeyCode::Char('k') => {
+                            app.previous();
+                        }
+                        KeyCode::Char('n') => {
+                            app.toggle_sort(SortColumn::Node);
+                        }
+                        KeyCode::Char('c') => {
+                            app.toggle_sort(SortColumn::Cpu);
+                        }
+                        KeyCode::Char('m') => {
+                            app.toggle_sort(SortColumn::Memory);
+                        }
+                        _ => {}
+                    }
+                }
+            }
+        }
+
+        // Update data if refresh interval has passed
+        if last_update.elapsed() >= refresh_duration {
+            // Refresh system info
+            system.refresh_all();
+
+            // Get node info from coordinator
+            let mut session = connect_to_coordinator((coordinator_addr, coordinator_port).into())
+                .wrap_err("Failed to connect to coordinator")?;
+
+            let request = ControlRequest::GetNodeInfo;
+            let reply_raw = session
+                .request(&serde_json::to_vec(&request).unwrap())
+                .wrap_err("failed to send request to coordinator")?;
```

Comment:
Is the node info expected to change during calls? It looks like this will always return the same result, so querying it again and again in every loop iteration is just uneccessary overhead.